src/docker-utils.coffee

```diff
@@ -15,6 +17,72 @@ Promise.promisifyAll(docker.getContainer().constructor.prototype)
 
 exports.docker = docker
 
+# Create an array of (repoTag, image_id, created) tuples like the output of `docker images`
+listRepoTagsAsync = ->
+	docker.listImagesAsync()
+	.then (images) ->
+		images = _.sortByOrder(images, 'Created', [ false ])
+		ret = []
+		for image in images
+			for repoTag in image.RepoTags
+				ret.push [ repoTag, image.Id, image.Created ]
+		return ret
+
+# Find either the most recent image of the same app or the image of the supervisor
+findSimilarImage = (repoTag) ->
+	application = repoTag.split('/')[1]
+
+	listRepoTagsAsync()
+	.then (repoTags) ->
+		# Find the most recent image of the same application
+		for repoTag in repoTags
+			otherApplication = repoTag[0].split('/')[1]
+			if otherApplication is application
+				return repoTag
+
+		# Otherwise return the most specific supervisor tag (commit hash)
+		for repoTag in repoTags when /resin\/.*-supervisor.*:[0-9a-f]{6}/.test(repoTag)
```

Comment:
This can use `config.supervisorImage` in order to match whatever the supervisor image name might be